base/applications/mspaint/history.cpp

```diff
@@ -267,3 +267,47 @@ HBITMAP ImageModel::CopyBitmap()
     m_hbmOld = ::SelectObject(m_hDrawingDC, m_hBms[m_currInd]); // Re-select
     return ret;
 }
+
+BOOL ImageModel::IsBlackAndWhite()
+{
+    LONG cxWidth = GetWidth(), cyHeight = GetHeight();
+    for (LONG y = 0; y < cyHeight; ++y)
+    {
+        for (LONG x = 0; x < cxWidth; ++x)
+        {
+            COLORREF rgbColor = ::GetPixel(m_hDrawingDC, x, y);
+            if (rgbColor != RGB(0, 0, 0) && rgbColor != RGB(255, 255, 255))
+                return FALSE;
+        }
+    }
+    return TRUE;
+}
```

Comment:
I think this would be slow on big images (and method is invoked each time on attributes dialog).

Reply:
https://github.com/reactos/reactos/pull/5563